server/src/main/java/com/hedera/block/server/persistence/storage/read/BlockAsLocalFileReader.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.persistence.storage.read;
+
+import com.hedera.block.common.utils.Preconditions;
+import com.hedera.hapi.block.BlockUnparsed;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+
+/**
+ * A Block reader that reads block-as-file.
+ */
+public final class BlockAsLocalFileReader implements LocalBlockReader<BlockUnparsed> {
+    /**
+     * Constructor.
+     */
+    private BlockAsLocalFileReader() {}
+
+    /**
+     * This method creates and returns a new instance of
+     * {@link BlockAsLocalFileReader}.
+     *
+     * @return a new, fully initialized instance of
+     * {@link BlockAsLocalFileReader}
+     */
+    public static BlockAsLocalFileReader newInstance() {
```

Comment:
As mentioned in [this comment thread](https://github.com/hashgraph/hedera-block-node/pull/355#discussion_r1857402179), I have migrated all reader/writer/remover/resolver classes to use static `of` methods for instance creation. Purpose is we do not have any optional dependencies as of yet. Once these methods are used, it then becomes trivial to introduce dependencies as it is really easy to refactor. Indeed, very soon this method will have to receive external dependencies. I chose the method `newInstance` for the sole reason that I have no params as of yet and the name `of` and then passing nothing to it seems off :). But you are right to say that the name suggest that it would be used not as a `singleton`, where a method name `getInstance` would be much more appropriate. But because we rely on `dagger` to do our DI and instance management, and we do not have nested singleton instance holders, the method name `newInstance` as of today reflects best the current situation, which is, each invocation will produce a new instance.

So it will get renamed to `of` and be passed dependencies once the class requires them. And the reason as to why not every static method is named `newInstance` is because we have planned forward compatibility where if in the future we have optional dependencies for a given class, we shall introduce a `Joshua Bloch` style builder which will be returned by the `of` method. In that situation, the name `newInstance` does not fit, since we would expect a new instance instead of a builder. That's the whole reasoning, code is as simple as possible until we need actually the added complexity of the builders.